src/ledger/parse/utils.js

```diff
@@ -58,13 +58,32 @@ function isPartialPayment(tx) {
 function parseDeliveredAmount(tx: Object): Amount | void {
   let deliveredAmount;
 
-  // TODO: Workaround for existing rippled bug where delivered_amount may not be
-  // provided for account_tx
-  if (tx.TransactionType === 'Payment') {
-    if (tx.meta.delivered_amount) {
+  if (tx.TransactionType === 'Payment' &&
+      tx.meta.TransactionResult === 'tesSUCCESS') {
+
+    if (tx.meta.delivered_amount &&
+        tx.meta.delivered_amount === 'unavailable') {
+      deliveredAmount = 'unavailable';
+
+    // parsable delivered_amount
+    } else if (tx.meta.delivered_amount) {
       deliveredAmount = parseAmount(tx.meta.delivered_amount);
+
+    // DeliveredAmount only present on partial payments
+    } else if (tx.meta.DeliveredAmount) {
+      deliveredAmount = parseAmount(tx.meta.DeliveredAmount);
+
+    // no partial payment flag
     } else if (tx.Amount && !isPartialPayment(tx)) {
       deliveredAmount = parseAmount(tx.Amount);
+
+    // partial payment after 4594094 without DeliveredAmount
+    } else if (tx.Amount && tx.ledger_index > 4594094) {
+      deliveredAmount = parseAmount(tx.Amount);
```

Comment:
I think we need a comment for this, along with an explanation for what changed in ledger version 4594094